roles/openshift_health_checker/test/etcd_volume_test.py

```diff
@@ -0,0 +1,131 @@
+
+    for word in 'determine available disk'.split() + extra_words:
+        assert word in str(excinfo.value)
+
+
+@pytest.mark.parametrize('ansible_mounts', [
+    [{
+        'mount': '/',
+        'size_available': 40 * 10**9 + 1,
+        'size_total': 80 * 10**9 + 1
+    }],
+    [{
+        'mount': '/',
+        'size_available': 30 * 10**9,
+        'size_total': 30 * 10**9 + 1,
+    }],
+    [{
+        'mount': '/',
+        'size_available': 20 * 10**9 + 1,
+        'size_total': 40 * 10**9 + 1,
+    }],
+    [{
+        # not enough space on / ...
+        'mount': '/',
+        'size_available': 0,
+        'size_total': 0,
+    }, {
+        # not enough space on /var/lib ...
+        'mount': '/var/lib',
+        'size_available': 2 * 10**9 + 1,
+        'size_total': 21 * 10**9 + 1,
+    }, {
+        # ... but enough on /var/lib/etcd
+        'mount': '/var/lib/etcd',
+        'size_available': 36 * 10**9 + 1,
+        'size_total': 40 * 10**9
+    }],
+])
+def test_succeeds_with_recommended_disk_space(ansible_mounts):
+    task_vars = dict(
+        ansible_mounts=ansible_mounts,
+    )
+
+    check = EtcdVolume(execute_module=fake_execute_module)
+    result = check.run(tmp=None, task_vars=task_vars)
+
+    assert not result.get('failed', False)
+
+
+@pytest.mark.parametrize('size_limit,ansible_mounts,extra_words', [
+    (
+        # if no size limit is specified, expect max usage
+        # limit to default to 90% of size_total
+        None,
+        [{
+            'mount': '/',
+            'size_available': 1 * 10**9 + 1,
+            'size_total': 100 * 10**9 + 1,
+        }],
+        ['90.00 GB'],
+    ),
+    (
+        3000000000,
+        [{
+            'mount': '/',
+            'size_available': 1 * 10**6,
+            'size_total': 5 * 10**9,
+        }],
+        ['3.00 GB'],
+    ),
+    (
+        3000000000,
+        [{
+            'mount': '/',
+            'size_available': 2 * 10**9,
+            'size_total': 5 * 10**9 + 1,
+        }],
+        ['3.00 GB'],
+    ),
+    (
+        None,
+        [{
+            # enough space on /var ...
+            'mount': '/var',
+            'size_available': 20 * 10**9,
+            'size_total': 20 * 10**9 + 1
+        }, {
+            # .. but not enough on /var/lib
+            'mount': '/var/lib',
+            'size_available': 1 * 10**9,
+            'size_total': 20 * 10**9,
+        }],
+        ['18.00 GB'],
+    ),
+])
+def test_fails_with_insufficient_disk_space(size_limit, ansible_mounts, extra_words):
+    task_vars = dict(
+        etcd_disk_size_limit_bytes=size_limit,
+        ansible_mounts=ansible_mounts,
+    )
+
+    if task_vars["etcd_disk_size_limit_bytes"] is None:
+        task_vars.pop("etcd_disk_size_limit_bytes", None)
+
+    check = EtcdVolume(execute_module=fake_execute_module)
+    result = check.run(tmp=None, task_vars=task_vars)
+
+    assert result['failed']
+    for word in 'greater than the recommended'.split() + extra_words:
```

Comment:
Hmm, looks like a misuse of a pattern here :)

This will look for the strings "greater", "than", "the" and "recommended" in no particular order (and not necessarily individual words, e.g., "there" contains "the"). I don't see how good it is to look for a generic string such as "the" in the output. Sure not harmful, but perhaps intention and actual code behavior may not match.